mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+                Error = data.Error
+                Ncount = data.Ncount
+
+                cut_max = data.plot_options.cut_max  # Default 1
+                cut_min = data.plot_options.cut_min  # Default 0
+
+                if data.plot_options.log:
+                    to_plot = Intensity
+
+                    max_data_value = to_plot.max()
+
+                    min_value = np.min(Intensity[np.nonzero(Intensity)])
+                    min_value = np.log10(min_value
+                                         + (max_data_value-min_value)*cut_min)
+
+                    max_value = np.log10(max_data_value*cut_max)
+
+                    if (max_value - min_value
+                            > data.plot_options.orders_of_mag):
+                        min_value = (max_value
+                                     - data.plot_options.orders_of_mag)
+                    min_value = 10.0 ** min_value
+                    max_value = 10.0 ** max_value
+                else:
+                    to_plot = Intensity
+                    min_value = to_plot.min() 
+                    max_value = to_plot.max()
+
+                    # Cut top and bottom of data as specified in cut variables
+                    min_value = min_value + (max_value-min_value)*cut_min
+                    max_value = max_value*cut_max
+
+                # Check the size of the array to be plotted
+                # print(to_plot.shape)
+
+                # Set the axis (might be switched?)
+                x_axis_mult = data.plot_options.x_limit_multiplier
+                y_axis_mult = data.plot_options.y_limit_multiplier
+
+                X = np.linspace(data.metadata.limits[0]*x_axis_mult,
+                                data.metadata.limits[1]*x_axis_mult,
+                                data.metadata.dimension[0]+1)
+                Y = np.linspace(data.metadata.limits[2]*y_axis_mult,
+                                data.metadata.limits[3]*y_axis_mult,
+                                data.metadata.dimension[1]+1)
+
+                # Create a meshgrid for both x and y
+                x, y = np.meshgrid(X, Y)
+
+                # Generate information on necessary colorrange
+                levels = MaxNLocator(nbins=150).tick_values(min_value,
+                                                            max_value)
+
+                # Select colormap
+                cmap = plt.get_cmap('hot')
+                norm = BoundaryNorm(levels, ncolors=cmap.N, clip=True)
+
+                # Create the figure
+                fig, (ax0) = plt.subplots()
+
+                # Plot the data on the meshgrids
+                if data.plot_options.log:
+                    color_norm = matplotlib.colors.LogNorm(vmin=min_value,
+                                                           vmax=max_value)
+                    im = ax0.pcolormesh(x, y, to_plot,
+                                        cmap=cmap, norm=color_norm)
+                else:
```

Comment:
you could make this shorter with something like
```Py
if data.plot_options.log:
    norm = matplotlib.colors.LogNorm(vmin=min_value,
                                                           vmax=max_value)
else:
    norm = BoundaryNorm(levels, ncolors=cmap.N, clip=True)
im = ax0.pcolormesh(x, y, to_plot, cmap=cmap, norm=norm)
```
i don't think `norm` is used anywhere else below, so should be ok to do this